Add Database Logsink Resources

do-joe opened the pull request:
## Changes

- New `digitalocean_database_logsink_rsyslog` resource for forwarding database logs to rsyslog servers
- New `digitalocean_database_logsink_opensearch` resource for forwarding database logs to OpenSearch/Elasticsearch clusters
- Full CRUD operations and import support for both resources
- Comprehensive validation (port ranges, formats, HTTPS enforcement, cross-field dependencies)
- Security features (TLS/mTLS for rsyslog, HTTPS-only for OpenSearch, sensitive field handling)
- Documentation for both resources with multiple configuration examples
- Updated database example to include rsyslog logsink configuration
- Engine compatibility: PostgreSQL, MySQL, Kafka, Valkey (MongoDB uses different mechanism)
- Update Godo to latest as bug fix in https://github.com/digitalocean/godo/pull/928 is needed

## Motivation

These resources allow users to aggregate and analyze database logs using their existing logging infrastructure. Users can forward logs to rsyslog servers (with optional TLS/mTLS) or OpenSearch/Elasticsearch clusters for centralized log management, monitoring, and analysis.

## Testing

- **Rsyslog acceptance tests:** 10/10 passing
  - `TestAccDigitalOceanDatabaseLogsinkRsyslog_Basic`
  - `TestAccDigitalOceanDatabaseLogsinkRsyslog_Update`
  - `TestAccDigitalOceanDatabaseLogsinkRsyslog_CustomFormat`
  - `TestAccDigitalOceanDatabaseLogsinkRsyslog_TLS`
  - `TestAccDigitalOceanDatabaseLogsinkRsyslog_ImportBasic`
  - `TestAccDigitalOceanDatabaseLogsinkRsyslog_InvalidPort`
  - `TestAccDigitalOceanDatabaseLogsinkRsyslog_InvalidFormat`
  - `TestAccDigitalOceanDatabaseLogsinkRsyslog_CustomFormatRequiresLogline`
  - `TestAccDigitalOceanDatabaseLogsinkRsyslog_CertWithoutTLS`
  - `TestAccDigitalOceanDatabaseLogsinkRsyslog_MongoDB_ShouldFail`

- **OpenSearch acceptance tests:** 7/7 passing
  - `TestAccDigitalOceanDatabaseLogsinkOpensearch_Basic`
  - `TestAccDigitalOceanDatabaseLogsinkOpensearch_Update`
  - `TestAccDigitalOceanDatabaseLogsinkOpensearch_WithCA`
  - `TestAccDigitalOceanDatabaseLogsinkOpensearch_ImportBasic`
  - `TestAccDigitalOceanDatabaseLogsinkOpensearch_InvalidIndexDaysMax`
  - `TestAccDigitalOceanDatabaseLogsinkOpensearch_EmptyIndexPrefix`
  - `TestAccDigitalOceanDatabaseLogsinkOpensearch_MalformedEndpoint`

anup-deka commented:
I can see there are changes in vendor files and no version updates in go.mod file
I would recommend running `go mod vendor` and `go mod tidy` to revert your vendor changes

do-joe commented:
@anup-deka thanks for catching that. I have removed the patched vendor code and rebased on the current main branch.  Tests still pass. CC @SSharma-10 

```
     === RUN   TestAccDigitalOceanDatabaseLogsinkRsyslog_ImportBasic                                 
     === PAUSE TestAccDigitalOceanDatabaseLogsinkRsyslog_ImportBasic
     === RUN   TestAccDigitalOceanDatabaseLogsinkOpensearch_ImportBasic
     === PAUSE TestAccDigitalOceanDatabaseLogsinkOpensearch_ImportBasic
     === RUN   TestAccDigitalOceanDatabaseLogsinkOpensearch_Basic
     === PAUSE TestAccDigitalOceanDatabaseLogsinkOpensearch_Basic
     === RUN   TestAccDigitalOceanDatabaseLogsinkOpensearch_Update
     === PAUSE TestAccDigitalOceanDatabaseLogsinkOpensearch_Update
     === RUN   TestAccDigitalOceanDatabaseLogsinkOpensearch_WithCA
     === PAUSE TestAccDigitalOceanDatabaseLogsinkOpensearch_WithCA
     === RUN   TestAccDigitalOceanDatabaseLogsinkOpensearch_InvalidIndexDaysMax
     === PAUSE TestAccDigitalOceanDatabaseLogsinkOpensearch_InvalidIndexDaysMax
     === RUN   TestAccDigitalOceanDatabaseLogsinkOpensearch_EmptyIndexPrefix
     === PAUSE TestAccDigitalOceanDatabaseLogsinkOpensearch_EmptyIndexPrefix
     === RUN   TestAccDigitalOceanDatabaseLogsinkOpensearch_MalformedEndpoint
     === PAUSE TestAccDigitalOceanDatabaseLogsinkOpensearch_MalformedEndpoint
     === RUN   TestAccDigitalOceanDatabaseLogsinkRsyslog_Basic
     === PAUSE TestAccDigitalOceanDatabaseLogsinkRsyslog_Basic
     === RUN   TestAccDigitalOceanDatabaseLogsinkRsyslog_Update
     === PAUSE TestAccDigitalOceanDatabaseLogsinkRsyslog_Update
     === RUN   TestAccDigitalOceanDatabaseLogsinkRsyslog_CustomFormat
     === PAUSE TestAccDigitalOceanDatabaseLogsinkRsyslog_CustomFormat
     === RUN   TestAccDigitalOceanDatabaseLogsinkRsyslog_TLS
     === PAUSE TestAccDigitalOceanDatabaseLogsinkRsyslog_TLS
     === RUN   TestAccDigitalOceanDatabaseLogsinkRsyslog_MongoDB_ShouldFail
     === PAUSE TestAccDigitalOceanDatabaseLogsinkRsyslog_MongoDB_ShouldFail
     === RUN   TestAccDigitalOceanDatabaseLogsinkRsyslog_InvalidPort
     === PAUSE TestAccDigitalOceanDatabaseLogsinkRsyslog_InvalidPort
     === RUN   TestAccDigitalOceanDatabaseLogsinkRsyslog_CustomFormatRequiresLogline
     === PAUSE TestAccDigitalOceanDatabaseLogsinkRsyslog_CustomFormatRequiresLogline
     === RUN   TestAccDigitalOceanDatabaseLogsinkRsyslog_InvalidFormat
     === PAUSE TestAccDigitalOceanDatabaseLogsinkRsyslog_InvalidFormat
     === RUN   TestAccDigitalOceanDatabaseLogsinkRsyslog_CertWithoutTLS
     === PAUSE TestAccDigitalOceanDatabaseLogsinkRsyslog_CertWithoutTLS
     === CONT  TestAccDigitalOceanDatabaseLogsinkRsyslog_ImportBasic
     === CONT  TestAccDigitalOceanDatabaseLogsinkRsyslog_Update
     === CONT  TestAccDigitalOceanDatabaseLogsinkRsyslog_InvalidFormat
     === CONT  TestAccDigitalOceanDatabaseLogsinkRsyslog_MongoDB_ShouldFail
     === CONT  TestAccDigitalOceanDatabaseLogsinkRsyslog_TLS
     === CONT  TestAccDigitalOceanDatabaseLogsinkRsyslog_CustomFormat
     === CONT  TestAccDigitalOceanDatabaseLogsinkRsyslog_InvalidPort
     === CONT  TestAccDigitalOceanDatabaseLogsinkRsyslog_CustomFormatRequiresLogline
     === CONT  TestAccDigitalOceanDatabaseLogsinkOpensearch_InvalidIndexDaysMax
     === CONT  TestAccDigitalOceanDatabaseLogsinkRsyslog_CertWithoutTLS
     === CONT  TestAccDigitalOceanDatabaseLogsinkRsyslog_Basic
     === CONT  TestAccDigitalOceanDatabaseLogsinkOpensearch_MalformedEndpoint
     --- PASS: TestAccDigitalOceanDatabaseLogsinkRsyslog_InvalidFormat (0.77s)
     === CONT  TestAccDigitalOceanDatabaseLogsinkOpensearch_Update
     --- PASS: TestAccDigitalOceanDatabaseLogsinkOpensearch_MalformedEndpoint (0.80s)
     === CONT  TestAccDigitalOceanDatabaseLogsinkOpensearch_WithCA
     --- PASS: TestAccDigitalOceanDatabaseLogsinkOpensearch_InvalidIndexDaysMax (0.82s)
     === CONT  TestAccDigitalOceanDatabaseLogsinkOpensearch_EmptyIndexPrefix
     --- PASS: TestAccDigitalOceanDatabaseLogsinkRsyslog_CustomFormatRequiresLogline (0.82s)
     === CONT  TestAccDigitalOceanDatabaseLogsinkOpensearch_Basic
     --- PASS: TestAccDigitalOceanDatabaseLogsinkRsyslog_InvalidPort (0.83s)
     === CONT  TestAccDigitalOceanDatabaseLogsinkOpensearch_ImportBasic
     --- PASS: TestAccDigitalOceanDatabaseLogsinkRsyslog_CertWithoutTLS (0.86s)
     --- PASS: TestAccDigitalOceanDatabaseLogsinkOpensearch_EmptyIndexPrefix (0.36s)
     --- PASS: TestAccDigitalOceanDatabaseLogsinkOpensearch_Basic (280.56s)
     --- PASS: TestAccDigitalOceanDatabaseLogsinkRsyslog_CustomFormat (281.58s)
     --- PASS: TestAccDigitalOceanDatabaseLogsinkRsyslog_ImportBasic (297.33s)
     --- PASS: TestAccDigitalOceanDatabaseLogsinkOpensearch_Update (300.37s)
     --- PASS: TestAccDigitalOceanDatabaseLogsinkOpensearch_WithCA (310.60s)
     --- PASS: TestAccDigitalOceanDatabaseLogsinkOpensearch_ImportBasic (311.27s)
     --- PASS: TestAccDigitalOceanDatabaseLogsinkRsyslog_Update (317.42s)
     --- PASS: TestAccDigitalOceanDatabaseLogsinkRsyslog_Basic (325.84s)
     --- PASS: TestAccDigitalOceanDatabaseLogsinkRsyslog_TLS (355.35s)
     --- PASS: TestAccDigitalOceanDatabaseLogsinkRsyslog_MongoDB_ShouldFail (591.02s)
     PASS
     ok         github.com/digitalocean/terraform-provider-digitalocean/digitalocean/database   591.037s
```
